include/openrave/geometry.h

```diff
@@ -2002,125 +1996,111 @@ inline bool AABBCollision(const RaveAxisAlignedBox<T>& ab1, const RaveAxisAligne
-//
-//	// A2 x B0 (-b0.y, b0.x, 0)
-//	r01 = a.vExtents.x * fabsf(o.vRight.y) + a.vExtents.y * fabsf(o.vRight.x) +
-//		o.vExtents.y * fabsf(o.vDir.z) + o.vExtents.z * fabsf(o.vUp.z);
-//	r = fabsf( o.vRight.x * vd.y - o.vRight.y * vd.x);
-//	if(r01 < r) return false;
-//
-//	// A2 x B1 (-b1.y, b1.x, 0)
-//	r01 = a.vExtents.x * fabsf(o.vUp.y) + a.vExtents.y * fabsf(o.vUp.x) +
-//		o.vExtents.x * fabsf(o.vDir.z) + o.vExtents.z * fabsf(o.vRight.z);
-//	r = fabsf( o.vUp.x * vd.y - o.vUp.y * vd.x);
-//	if(r01 < r) return false;
-//
-//	// A2 x B2 (-b2.y, b2.x, 0)
-//	r01 = a.vExtents.x * fabsf(o.vDir.y) + a.vExtents.y * fabsf(o.vDir.x) +
-//		o.vExtents.x * fabsf(o.vUp.z) + o.vExtents.y * fabsf(o.vRight.z);
-//	r = fabsf( o.vDir.x * vd.y - o.vDir.y * vd.x);
-//	if(r01 < r) return false;
-//
-//	return true;
-//}
-//
-//bool OBBOBBTest(const OBB& o1, const OBB& o2)
-//{
-//	// convert to AABB and OBB
-//	AABB ab(DXVEC3(0.0f, 0.0f, 0.0f), o1.vExtents);
-//	OBB obb;
-//
-//	// position
-//	ab.vPos = o2.vPos - o1.vPos;
-//	obb.vPos.x = D3DXVec3Dot(&o1.vRight, &ab.vPos);
-//	obb.vPos.y = D3DXVec3Dot(&o1.vUp, &ab.vPos);
-//	obb.vPos.z = D3DXVec3Dot(&o1.vDir, &ab.vPos);
-//	ab.vPos.x = ab.vPos.y = ab.vPos.z = 0.0f;
-//
-//	obb.vRight.x = D3DXVec3Dot(&o1.vRight, &o2.vRight);
-//	obb.vRight.y = D3DXVec3Dot(&o1.vUp, &o2.vRight);
-//	obb.vRight.z = D3DXVec3Dot(&o1.vDir, &o2.vRight);
-//
-//	obb.vUp.x = D3DXVec3Dot(&o1.vRight, &o2.vUp);
-//	obb.vUp.y = D3DXVec3Dot(&o1.vUp, &o2.vUp);
-//	obb.vUp.z = D3DXVec3Dot(&o1.vDir, &o2.vUp);
-//
-//	obb.vDir.x = D3DXVec3Dot(&o1.vRight, &o2.vDir);
-//	obb.vDir.y = D3DXVec3Dot(&o1.vUp, &o2.vDir);
-//	obb.vDir.z = D3DXVec3Dot(&o1.vDir, &o2.vDir);
-//
-//	obb.vExtents = o2.vExtents;
-//
-//	return AABBOBBTest(ab, obb);
-//}
+/// \brief Test collision between two oriented bounding boxes.
+template <typename T>
+inline bool OBBCollision(const RaveOrientedBox<T>& obb1, const RaveOrientedBox<T>& obb2)
+{
+    const RaveTransform<T> relativeTransform = obb1.transform.inverse() * obb2.transform;
+    return BoxAtOriginOBBCollision(obb1.extents, RaveOrientedBox<T>(relativeTransform, obb2.extents));
+}
+
+/// \brief Test collision between an axis-aligned bounding box and an oriented bounding box.
+template <typename T>
+inline bool AABBOBBCollision(const RaveAxisAlignedBox<T>& ab, const RaveOrientedBox<T>& obb)
+{
+    const RaveTransform<T> relativeTransform(obb.transform.rot, obb.transform.trans - ab.pos);
+    return BoxAtOriginOBBCollision(ab.extents, RaveOrientedBox<T>(relativeTransform, obb.extents));
+}
+
+/// \brief Test collision between an axis-aligned bounding box located at world origin and an oriented bounding box.
+/// using Separating Axis Theorem.
+/// S. Gottschalk, M.C. Lin, D. Manocha. OBBTree: A Hierarchical Structure for Rapid Interference Detection. In Proceedings of SIGGRAPH '96.
+template <typename T>
+inline bool BoxAtOriginOBBCollision(const RaveVector<T>& extents, const RaveOrientedBox<T>& obb)
```

Comment:
Given these are new functions, perhaps you can give them better names, ones that start with a verb like Check?

Reply:
@rdiankov Thank you, improved!

Reply:
@rdiankov i think it's good again now.